### PR TITLE
i18n(fr) Up and Add french translation for `upgrade-astro.mdx` and `v5.mdx`

### PR DESCRIPTION
**Pull request by McFlyPartages**
#### Description (required)
Tranlation for french doc

Update `upgrade-astro` file and add `v5` astro file

### CODE REVIEW

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 1]
```diff
@@ -0,0 +1,1205 @@
...
+{
+  "extends": "astro/tsconfigs/base",
+  "include": [".astro/types.d.ts", "**/*"],
+  "exclude": ["dist"]
+}
+```
+
+Notez que `src/env.d.ts` n'est nécessaire que si vous avez ajouté des configurations personnalisées, ou si vous n'utilisez pas un fichier `tsconfig.json`.
+
+<ReadMore>En savoir plus sur [TypeScript configuration in Astro](/fr/guides/typescript/#configuration).</ReadMore>
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:602]:
```suggestion
<ReadMore>En savoir plus sur la [configuration de TypeScript dans Astro](/fr/guides/typescript/#configuration).</ReadMore>
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 2]
```diff
@@ -0,0 +1,1205 @@
...
+
+Astro 5.0 rend maintenant le résultat d'une action comme un résultat POST sans aucune redirection. Cela introduira une boîte de dialogue « confirmer la resoumission du formulaire ? » lorsqu'un utilisateur tentera de rafraîchir la page, bien que cela n'impose plus une limite de 4 Ko sur la valeur de retour de l'action.
+
+#### Que dois-je faire ?
+
+Vous devez mettre à jour le traitement des résultats des actions qui reposent sur des redirections, et éventuellement traiter la boîte de dialogue « confirmer la resoumission du formulaire ? » avec un middleware.
+
+##### Rediriger vers la route précédente en cas d'erreur
+
+Si l'action de votre formulaire HTML est dirigée vers une route différente (c'est-à-dire `action={« /success-page » + actions.name}`), Astro ne redirigera plus vers la route précédente en cas d'erreur. Vous pouvez implémenter ce comportement manuellement en utilisant les redirections de votre composant Astro. Cet exemple redirige plutôt vers une nouvelle route en cas de succès, et gère les erreurs sur la page courante dans le cas contraire :
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:618]:
```suggestion
Si l'action de votre formulaire HTML est dirigée vers une route différente (c'est-à-dire `action={"/success-page" + actions.name}`), Astro ne redirigera plus vers la route précédente en cas d'erreur. Vous pouvez implémenter ce comportement manuellement en utilisant les redirections de votre composant Astro. Cet exemple redirige plutôt vers une nouvelle route en cas de succès, et gère les erreurs sur la page courante dans le cas contraire :
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 3]
```diff
@@ -0,0 +1,1205 @@
...
+
+<form method="POST" action={'/confirmation' + actions.newsletter}>
+  <label>E-mail <input required type="email" name="email" /></label>
+  <button>Sign up</button>
+</form>
+```
+
+##### (Optionnel) Supprimer la boîte de dialogue de confirmation lors de l'actualisation
+
+Pour résoudre le problème de la boîte de dialogue « Confirmation de la resoumission du formulaire ? » lors de l'actualisation, ou pour conserver les résultats des actions entre les sessions, vous pouvez désormais [personnaliser la gestion des résultats des actions à partir d'un middleware'](/fr/guides/actions/#avancé-persistance-des-résultats-daction-avec-une-session).
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:640]:
```suggestion
Pour résoudre le problème de la boîte de dialogue « Confirmation de la resoumission du formulaire ? » lors de l'actualisation, ou pour conserver les résultats des actions entre les sessions, vous pouvez désormais [personnaliser la gestion des résultats des actions à partir d'un middleware](/fr/guides/actions/#avancé-persistance-des-résultats-daction-avec-une-session).
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 4]
```diff
@@ -0,0 +1,1205 @@
...
+
+<ReadMore>En savoir plus sur [l'API `astro:content`](/fr/reference/modules/astro-content/).</ReadMore>
+
+### Renommé : Noms des jetons de couleur du thème Shiki `css-variables`.
+
+<SourcePR number="11661" title="Mise à jour vers les nouveaux noms de jetons shiki"/>
+
+Dans Astro v4.x, le thème Shiki `css-variables` utilisait les jetons `--astro-code-color-text` et `--astro-code-color-background` pour styliser les couleurs d'avant-plan et d'arrière-plan des blocs de code respectivement.
+
+Astro v5.0 les renomme respectivement en `--astro-code-foreground` et `--astro-code-background` pour mieux s'aligner sur les valeurs par défaut de Shiki v1..
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:747]:
```suggestion
Astro v5.0 les renomme respectivement en `--astro-code-foreground` et `--astro-code-background` pour mieux s'aligner sur les valeurs par défaut de Shiki v1.
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 5]
```diff
@@ -0,0 +1,1205 @@
...
+
+<ReadMore>Plus d'informations sur [la coloration syntaxique dans Astro](/fr/guides/syntax-highlighting/).</ReadMore>
+
+### Modifié : Comportement automatique `charset=utf-8` pour les pages Markdown et MDX
+
+<SourcePR number="12231" title="Type de contenu non défini charset=utf-8 pour les pages md/mdx"/>
+
+Dans Astro 4.0, les pages Markdown et MDX (situées dans `src/pages/`) répondaient automatiquement avec `charset=utf-8` dans l'en-tête `Content-Type`, ce qui permettait de rendre les caractères non-ASCII dans vos pages.
+
+Astro 5.0 met à jour ce comportement en ajoutant la balise `<meta charset=« utf-8 »>` à la place, et seulement pour les pages qui n'utilisent pas la propriété spéciale `layout` dans le frontmatter d'Astro. De même pour les pages MDX, Astro n'ajoutera la balise que si le contenu MDX n'importe pas un composant `Layout`.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:786]:
```suggestion
Astro 5.0 met à jour ce comportement en ajoutant la balise `<meta charset="utf-8">` à la place, et seulement pour les pages qui n'utilisent pas la propriété spéciale `layout` dans le frontmatter d'Astro. De même pour les pages MDX, Astro n'ajoutera la balise que si le contenu MDX n'importe pas un composant `Layout`.
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 6]
```diff
@@ -0,0 +1,1205 @@
...
+
+Si vos pages Markdown ou MDX utilisent la propriété `layout` dans le frontmatter, ou si le contenu de la page MDX importe un composant `Layout`, alors l'encodage HTML sera géré par le composant layout désigné à la place, et la balise `<meta charset=« utf-8 »>` ne sera pas ajoutée à votre page par défaut.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:788]:
```suggestion
Si vos pages Markdown ou MDX utilisent la propriété `layout` dans le frontmatter, ou si le contenu de la page MDX importe un composant `Layout`, alors l'encodage HTML sera géré par le composant layout désigné à la place, et la balise `<meta charset="utf-8">` ne sera pas ajoutée à votre page par défaut.
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 7]
```diff
@@ -0,0 +1,1205 @@
...
+
+#### Que dois-je faire ?
+
+Si vous avez besoin de `charset=utf-8` pour rendre votre page correctement, assurez-vous que vos composants de mise en page contiennent la balise `<meta charset=« utf-8 »>`. Vous devrez peut-être l'ajouter si vous ne l'avez pas encore fait.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:792]:
```suggestion
Si vous avez besoin de `charset=utf-8` pour rendre votre page correctement, assurez-vous que vos composants de mise en page contiennent la balise `<meta charset="utf-8">`. Vous devrez peut-être l'ajouter si vous ne l'avez pas encore fait.
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 8]
```diff
@@ -0,0 +1,1205 @@
...
+
+Ce changement a été fait parce que le comportement précédent a causé de la confusion parmi les auteurs d'intégration qui ont testé un paquet qui fonctionnait localement, mais pas lorsqu'il était publié. Cela a également limité l'utilisation des dépendances CJS uniquement parce que Vite exigeait que le code soit ESM. Bien que ce changement n'affecte que les dépendances JS, il est également recommandé pour les paquets d'exporter JavaScript au lieu de TypeScript brut lorsque c'est possible afin d'éviter une utilisation accidentelle spécifique à Vite car c'est un détail d'implémentation du flux de chargement de la configuration d'Astro.
+
+#### Que dois-je faire ?
+
+Assurez-vous que vos dépendances JS liées localement sont construites avant de lancer votre projet Astro. Ensuite, le chargement de la configuration devrait fonctionner comme avant.
+
+<ReadMore>En savoir plus sur [Les paramètres de configuration de Vite dans Astro](/fr/reference/configuration-reference/#vite).</ReadMore>
+
+### Changed: URLs returned by `paginate()`
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:881]:
```suggestion
### Modifié : les URL renvoyées par `paginate()`
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 9]
```diff
@@ -0,0 +1,1205 @@
...
+
+<ReadMore>En savoir plus sur [pagination dans Astro](/fr/guides/routing/#pagination).</ReadMore>
+
+### Modifié : valeurs d'attribut HTML non booléennes
+
+<SourcePR number="11660" title="Correction du rendu des attributs pour les valeurs booléennes (take 2)"/>
+
+Dans Astro v4.x, les attributs HTML non [booléens] (https://developer.mozilla.org/fr-FR/docs/Glossary/Boolean/HTML) pouvaient ne pas inclure leurs valeurs lorsqu'ils étaient rendus en HTML.
+
+Astro v5.0 rend les valeurs explicites comme `=« true »` ou `=« false »`, ce qui correspond à la gestion correcte des attributs dans les navigateurs.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:923]:
```suggestion
Astro v5.0 rend les valeurs explicites comme `="true"` ou `="false"`, ce qui correspond à la gestion correcte des attributs dans les navigateurs.
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 10]
```diff
@@ -0,0 +1,1205 @@
...
+```js del={1,4} ins={2,5}
+import type {RouteData} from 'astro';
+import type {IntegrationRouteData} from "astro"
+
+function useRoute(route: RouteData) {
+function useRoute(route: IntegrationRouteData) {
+}
+```
+
+<ReadMore>Voir la [Référence API pour `IntegrationRouteData`](/fr/reference/integrations-reference/#référence-du-type-integrationroutedata).</ReadMore>
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:1044]:
```suggestion
<ReadMore>Voir la [référence de l'API pour `IntegrationRouteData`](/fr/reference/integrations-reference/#référence-du-type-integrationroutedata).</ReadMore>
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 11]
```diff
@@ -0,0 +1,1205 @@
...
+  }
+  for (const url of route.distURL) {
+    if (url.endsWith('index.html')) {
+      // do something
+    }
+  }
+}
+```
+
+<ReadMore>Voir la [Référence API pour `IntegrationRouteData`](/fr/reference/integrations-reference/#référence-du-type-integrationroutedata).</ReadMore>
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:1071]:
```suggestion
<ReadMore>Voir la [référence de l'API pour `IntegrationRouteData`](/fr/reference/integrations-reference/#référence-du-type-integrationroutedata).</ReadMore>
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 12]
```diff
@@ -0,0 +1,1205 @@
...
+  ```diff lang=shell
+  -yarn create astro --template <example-name> --typescript strict
+  +yarn create astro --template <example-name>
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+<ReadMore>Voir [tous les paramètres de la commande `create astro` disponibles](https://github.com/withastro/astro/blob/main/packages/create-astro/README.md)</ReadMore>
+
+## Community Resources
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:1199]:
```suggestion
## Ressources communautaires
```

**Review by ArmandPhilippot — CHANGES_REQUESTED** [review]
Thanks for tackling this big file! I suggested some changes:
* mostly nitpicks (missing space, double dot...)
* some untranslated strings
* some malformed inline codes
* some malformed links

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 13]
```diff
@@ -0,0 +1,1205 @@
...
+<a id="prev" href={prev}>Back</a>
+```
+
+<ReadMore>En savoir plus sur [pagination dans Astro](/fr/guides/routing/#pagination).</ReadMore>
+
+### Modifié : valeurs d'attribut HTML non booléennes
+
+<SourcePR number="11660" title="Correction du rendu des attributs pour les valeurs booléennes (take 2)"/>
+
+Dans Astro v4.x, les attributs HTML non [booléens] (https://developer.mozilla.org/fr-FR/docs/Glossary/Boolean/HTML) pouvaient ne pas inclure leurs valeurs lorsqu'ils étaient rendus en HTML.
```
[the remark this answers is not in the document]
Comment (ArmandPhilippot) [reply src/content/docs/fr/guides/upgrade-to/v5.mdx:921]: Oh good catch, I missed that one!

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 14]
```diff
@@ -0,0 +1,1205 @@
...
+Après avoir mis à jour Astro, il est possible que vous n'ayez pas besoin d'apporter de modifications à votre projet !
+
+Cependant, si vous remarquez des erreurs ou un comportement inattendu, veuillez vérifier ci-dessous ce qui a changé et qui pourrait nécessiter une mise à jour dans votre projet.
+:::
+
+Astro v5.0 inclut des [changements potentiellement cassants](#changements-de-rupture), ainsi que la suppression et la dépréciation de certaines fonctionnalités.
+
+Si votre projet ne fonctionne pas comme prévu après la mise à jour vers la version 5.0, consultez ce guide pour un aperçu de tous les changements cassants et des instructions sur la façon de mettre à jour votre base de code.
+
+Consultez le changelog d'Astro](https://github.com/withastro/astro/blob/main/packages/astro/CHANGELOG.md) pour les notes de version complètes.
```
[the remark this answers is not in the document]
Comment (McFlyPartages) [reply src/content/docs/fr/guides/upgrade-to/v5.mdx]: Thanks i see

[the remark this answers is not in the document]
Comment (McFlyPartages) [reply src/content/docs/fr/guides/upgrade-to/v5.mdx]: Thanks to you for the review, it takes as long as translating hahaha

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 15]
```diff
@@ -0,0 +1,1205 @@
...
+
+#### Que dois-je faire ?
+
+Nous recommandons de [convertir toutes les collections existantes](#mise-à-jour-des-collections-existantes) vers la nouvelle API Content Layer dès que possible et de créer toute nouvelle collection en utilisant l'API Content Layer.
+
+Si vous ne pouvez pas convertir vos collections, veuillez consulter [les changements incompatibles pour les collections héritées](#rupture-des-anciennes-collections-content-et-data) pour voir si vos collections existantes sont affectées et nécessitent une mise à jour.
+
+Si vous ne pouvez apporter aucune modification à vos collections pour le moment, vous pouvez [activer l'option `legacy.collections`](#activation-de-l-option-legacy.collections) qui vous permettra de conserver vos collections dans leur état actuel jusqu'à ce que le drapeau hérité ne soit plus pris en charge.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:107]:
```suggestion
Si vous ne pouvez apporter aucune modification à vos collections pour le moment, vous pouvez [activer l'option `legacy.collections`](#activation-de-loption-legacycollections) qui vous permettra de conserver vos collections dans leur état actuel jusqu'à ce que le drapeau hérité ne soit plus pris en charge.
```

File: src/content/docs/fr/reference/integrations-reference.mdx [hunk 16]
```diff
@@ -893,7 +893,7 @@ export default function myIntegration() {
 #### Option `routes`
 
 :::caution
-Cette propriété est obsolète depuis la version 5.0. Consultez le [guide de migration](/fr/guides/upgrade-to/v5/#deprecated-routes-on-astrobuilddone-hook-integration-api).
+Cette propriété est obsolète depuis la version 5.0. Consultez le [guide de migration](/fr/guides/upgrade-to/v5/#déprécié-routes-sur-le-hook-astrobuilddoneapi-dintégration).
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/reference/integrations-reference.mdx:896]:
```suggestion
Cette propriété est obsolète depuis la version 5.0. Consultez le [guide de migration](/fr/guides/upgrade-to/v5/#déprécié--routes-sur-le-hook-astrobuilddoneapi-dintégration).
```

**Review by ArmandPhilippot — CHANGES_REQUESTED** [review]
This should fix the broken links!

**Comment by McFlyPartages** [comment]
Hey thanks for your review. It's good now 

Thanks Thanks Thanks and good job

**Review by thomasbnt — APPROVED** [review]
LGTM!

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 17]
```diff
@@ -0,0 +1,1205 @@
...
+
+<SourcePR number="11741" title="Nettoyage du code JSX inutilisé"/>
+
+Dans Astro v4.x, Astro effectuait un traitement interne du JSX pour l'intégration @astrojs/mdx.
+
+Astro v5.0 transfère cette responsabilité de gestion et de rendu du JSX et MDX directement au package @astrojs/mdx. Cela signifie qu'Astro 5.0 n'est plus compatible avec les anciennes versions de l'intégration MDX.
+
+#### Que dois-je faire ?
+
+Si votre projet inclut des fichiers .mdx, vous devez mettre à jour @astrojs/mdx vers la dernière version (v4.0.0) afin que votre JSX soit correctement géré par l'intégration.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:80]: nit:
```suggestion
Si votre projet inclut des fichiers `.mdx`, vous devez mettre à jour @astrojs/mdx vers la dernière version (v4.0.0) afin que votre JSX soit correctement géré par l'intégration.
```

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 18]
```diff
@@ -0,0 +1,1205 @@
...
+
+Si vous ne pouvez pas convertir vos collections, veuillez consulter [les changements incompatibles pour les collections héritées](#rupture-des-anciennes-collections-content-et-data) pour voir si vos collections existantes sont affectées et nécessitent une mise à jour.
+
+Si vous ne pouvez apporter aucune modification à vos collections pour le moment, vous pouvez [activer l'option `legacy.collections`](#activation-de-loption-legacycollections) qui vous permettra de conserver vos collections dans leur état actuel jusqu'à ce que le drapeau hérité ne soit plus pris en charge.
+
+<ReadMore>En savoir plus sur la [mise à jour des collections de contenu](/fr/guides/content-collections/).</ReadMore>
+
+##### Mise à jour des collections existantes
+
+Voir les instructions ci-dessous pour mettre à jour une collection de contenu existante (`type : 'content'` ou `type : 'data'`) afin d'utiliser l'API de la couche de contenu.
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:113]: nit, inline code no space before colon:
```suggestion
Voir les instructions ci-dessous pour mettre à jour une collection de contenu existante (`type: 'content'` ou `type: 'data'`) afin d'utiliser l'API de la couche de contenu.
```

**Review by ArmandPhilippot — APPROVED** [review]
Two tiny nits, otherwise LGTM!

**Review by ArmandPhilippot — APPROVED** [review]
LGTM!

File: src/content/docs/fr/guides/upgrade-to/v5.mdx [hunk 19]
```diff
@@ -0,0 +1,1205 @@
...
+---
+
+{posts.map((post) => <li><a href={post.url}>{post.frontmatter.title}</a></li>)}
+```
+
+Le cas échéant, envisagez d'utiliser les [collections de contenu](/fr/guides/content-collections/) pour organiser votre contenu, qui disposent de leurs propres fonctions d'interrogation, plus récentes et plus performantes.
+
+Vous pouvez également envisager d'utiliser les paquets globaux de NPM, tels que [`fast-glob`](https://www.npmjs.com/package/fast-glob).
+
+<ReadMore>En savoir plus sur [l'importation de fichiers avec `import.meta.glob`](/fr/guides/imports/#astroglob).</ReadMore>
```
Comment (thomasbnt) [inline src/content/docs/fr/guides/upgrade-to/v5.mdx:229]:
```suggestion
<ReadMore>En savoir plus sur [l'importation de fichiers avec `import.meta.glob`](/fr/guides/imports/#importmetaglob).</ReadMore>
```

**Review by thomasbnt — CHANGES_REQUESTED** [review]
Miss one link #importmetaglob and LGTM!

**Review by thomasbnt — APPROVED** [review]
LGTM!

**Pull request merged**